tests/instancetype_test.go

```diff
@@ -655,6 +657,387 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
+			Eventually(func() bool {
+				var err error
+				pvc, err = virtClient.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(context.Background(), pvc.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return pvc.Status.Phase == k8sv1.ClaimBound
+			}, 180*time.Second, time.Second).Should(BeTrue())
+			return pvc
+		}
+
+		BeforeAll(func() {
+			if !libstorage.HasCDI() {
+				Skip("instance type and preference inferFromVolume tests require CDI to be installed providing the DataVolume and DataSource CRDs")
+			}
+
+			blockStorageClass, blockStorageClassExists = libstorage.GetRWOBlockStorageClass()
+			if !blockStorageClassExists {
+				Skip("Skip test when RWOBlock storage class is not present")
+			}
+
+			By("Creating a VirtualMachineInstancetype")
+			instancetype = newVirtualMachineInstancetype(nil)
+			instancetype.Name = instancetypeName
+			instancetype, err = virtClient.VirtualMachineInstancetype(util.NamespaceTestDefault).
+				Create(context.Background(), instancetype, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Creating a VirtualMachinePreference")
+			preference = newVirtualMachinePreference()
+			preference.Name = preferenceName
+			preference.Spec = instancetypev1alpha2.VirtualMachinePreferenceSpec{
+				CPU: &instancetypev1alpha2.CPUPreferences{
+					PreferredCPUTopology: instancetypev1alpha2.PreferCores,
+				},
+			}
+			preference, err = virtClient.VirtualMachinePreference(util.NamespaceTestDefault).
+				Create(context.Background(), preference, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		BeforeEach(func() {
+			By("Creating an annotated source PVC")
+			sourcePVC = libstorage.NewPVC(pvcSourceName, "1Gi", blockStorageClass)
+			sourcePVCVolumeBlockMode := k8sv1.PersistentVolumeBlock
+			sourcePVC.Spec.VolumeMode = &sourcePVCVolumeBlockMode
+			sourcePVC.Annotations = map[string]string{
+				instancetypeapi.DefaultInstancetypeAnnotation:     instancetype.Name,
+				instancetypeapi.DefaultInstancetypeKindAnnotation: instancetypeapi.SingularResourceName,
+				instancetypeapi.DefaultPreferenceAnnotation:       preference.Name,
+				instancetypeapi.DefaultPreferenceKindAnnotation:   instancetypeapi.SingularPreferenceResourceName,
+			}
+			sourcePVC, err = virtClient.CoreV1().PersistentVolumeClaims(util.NamespaceTestDefault).Create(context.Background(), sourcePVC, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			waitPVCBound(sourcePVC)
+
+			vm = &v1.VirtualMachine{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "vm-",
+					Namespace:    util.NamespaceTestDefault,
+				},
+				Spec: v1.VirtualMachineSpec{
+					Instancetype: &v1.InstancetypeMatcher{
+						InferFromVolume: inferFromVolumeName,
+					},
+					Preference: &v1.PreferenceMatcher{
+						InferFromVolume: inferFromVolumeName,
+					},
+					Template: &v1.VirtualMachineInstanceTemplateSpec{
```

Comment:
nit: It may not be necessary to define the empty Template here.

Reply:
All of the tests set `vm.spec.template.spec.volumes` so it is required.

Reply:
ACK